src/Displaz.jl

```diff
@@ -304,8 +312,8 @@ function plot3d(plotobj::DisplazWindow, position; color=[1,1,1], markersize=[0.1
     end
 
     # works for vectors only at this stage...
-    extra_fields = ((x -> (size(x[2]) == (nvertices,) || error("extra fields must be vectors of same length as number of points in position array") ;
-        (x[1], array_semantic, transpose(Float32.(x[2])))) for x = pairs(kwargs)))
+    extra_fields = ((x -> (size(x[2]) == (nvertices,)) || error("extra fields must be vectors of same length as number of points in position array") ;
+        (x[1], array_semantic, transpose(convert_values{x[1]}.(x[2]))) for x = pairs(kwargs)))
```

Comment:
I think this will be quite slow - things happening in the type domain should generally be statically known at compile time. You could just add an `if` statement here?

Also, the `extra_fields` function would be clearer in "long" form like `function extra_fields(x) ... end` (you can do nested `function`s for closures). I'm actually having trouble parsing it!

Reply:
Tried to not change too much :) - I also don't really like `|| error(...);` bit, but left it as well. I temporarily had it as a "map" to understand what's going on here, and don't like the `for` usage like this as well. I'll change this then to a more readable loop then later, stay tuned :) - Any ideas how to unit test? For now I can only promise you "it works on my machine".

On the "slowness", I didn't notice a slowdown for a 6M point cloud. I actually think this compiles, since "kwargs" is a Tuple, and the "x"'s are possibly even known at compile time, so it picks the right `convert_values{x[1]}` at compile time.

Or, more likely, this is not a per-point operation: it will pick a converter once (or twice, at least per "field", like "intensity" or "classification") and then broadcast it to all values, so you won't notice any difference in speed.

Thanks for having a look!

Reply:
It's not even a `function extra_fields` by the way, it is a closure, but very nightmarish with the mega-hacky `|| error();` bit. You can actually remove the whole `(size(x[2]) == (nvertices,)) || error("extra fields must be vectors of same length as number of points in position array") ;` part and you'll notice it's only: `extra_fields = ((x -> (x[1], array_semantic, transpose(convert_values{x[1]}.(x[2]))) for x = pairs(kwargs)))` which maps a tuple of pairs into a tuple of some other 3-element-tuple.

Reply:
I was suggesting rewriting the whole two lines to make them legible, that's all. (You can use nested `function` for closures, it's 100% equivalent). But yeah it's probably only constructing this type once per collection not once per point, so I doubt it's a huge performance problem, but you've got code spread out now where you could deal with it with an `if x[1] == :intensity ... elseif ... end` kind of thing, and remove the `convert_values` thing entirely.